### PR TITLE
detect-replace.c:Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by Rai0Catodic0**
Ticket: 4054

- [:white_check_mark:  ] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [:white_check_mark: ] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- modify detect-replace to use FAIL/PALL APIs

### CODE REVIEW

File: src/detect-replace.c [hunk 1]
```diff
@@ -613,19 +620,16 @@ static int DetectReplaceParseTest01(void)
     de_ctx->sig_list = SigInit(de_ctx,
                                "alert udp any any -> any any "
                                "(msg:\"test\"; content:\"doh\"; replace:\"; sid:238012;)");
-    if (de_ctx->sig_list != NULL) {
-        result = 0;
-        goto end;
-    }
+    FAIL_IF_NOT_NULL(de_ctx->sig_list);
```
Comment (Rai0Catodic0) [inline src/detect-replace.c:623]: is it ok to use the macro here? I believe the code after the end label won't run.

**Review by jufajardini — CHANGES_REQUESTED** [review]
Hello, thanks for this contribution. There are a few changes missing, here, I've added inline comments, and some general stuff:
- nit: can you update the copyright year? We try to update those when we change the files (`2011-2014` -> `2011-2022`)
- CI check fails clang checks. Please run the clang script (`./scripts/clang-format.sh branch`) on your branch to have clang do the required formatting changes. :)
- It seems that tests `DetectReplaceMatchTest09` to `DetectReplaceMatchTest14`
  were skipped - thinking that this may have been due to the question that you shared, now you have the answer :)
- The comments added to the test `DetectReplaceParseTest01()` apply to all other
  tests with a similar structure in this file, and all should be converted to the FAIL/PASS api -- we don't want any if statements in our unittests, one of the goals is to have them all turned into FAIL/PASS macros.
If in doubt, you may check the rest of the changes (and any feedback) in the PR I shared in one of the inline comments.

I'm not sure about `DetectReplaceLongPatternMatchTest()`, let's leave it as is for now, and then with the next PR we may ask Victor for feedback on that.

**Comment by jufajardini** [comment]
> Thank you for your answer, It helped a lot. I will apply those changes and finish what was missing.

Thanks! Also do check the other CI failure (commit-check) and see how if you manage to fix them :)

**Comment by Rai0Catodic0** [comment]
closing this in favour of a newer PR #8046

**Pull request closed without merging**